src/test/java/com/example/music_player/service/SourceServiceTest.java

```diff
@@ -51,105 +57,82 @@ public void setup() {
         testArray = "testByteArray".getBytes();
         testBoolean = false;
         sourceList = List.of(source1, source2);
+        inputStream = new ByteArrayInputStream(StandardCharsets.UTF_16.encode("testString").array());
     }
 
     @Test
-    void save() {
-
+    void saveWhenSourceDoNotExist() throws IOException {
+        when(sourceRepository.isExistByNameAndFileType(song.getName()
+                , mockMultipartFile.getContentType()))
+                .thenReturn(testBoolean);
```

Comment:
It is hard to keep in mind this testBoolean field. use local variable instead, or better use false or true right in thenReturn statement.